cellular_automata/langtons_ant.py

```diff
@@ -0,0 +1,110 @@
+"""
+Langton's ant
+
+@ https://en.wikipedia.org/wiki/Langton%27s_ant
+@ https://upload.wikimedia.org/wikipedia/commons/0/09/LangtonsAntAnimated.gif
+"""
+
+from functools import partial
+
+from matplotlib import pyplot as plt
+from matplotlib.animation import FuncAnimation
+
+WIDTH = 80
+HEIGHT = 80
+
+
+class LangtonsAnt:
+    """
+    Represents the main LangonsAnt algorithm.
+
+    >>> la = LangtonsAnt(2, 2)
+    >>> la.board
+    [[True, True], [True, True]]
+    >>> la.ant_position
+    (1, 1)
+    """
+
+    def __init__(self, width: int, height: int) -> None:
+        # Each square is either True or False where True is white and False is black
+        self.board = [[True] * width for _ in range(height)]
+        self.ant_position: tuple[int, int] = (width // 2, height // 2)
+
+        # Initially pointing left (similar to the the wikipedia image)
+        # (0 = 0° | 1 = 90° | 2 = 180 ° | 3 = 270°)
+        self.ant_direction: int = 3
+
+    def move_ant(self, axes: plt.Axes | None, display: bool, _frame: int) -> None:
```

Comment:
What does `_frame` do? It doesn't appear to be used anywhere

Reply:
It doesn't do anything - It's an argument that is passed in from `FuncAnimation`
The preceding `_` indicates that its unused